src/vivarium_nih_us_cvd/components/healthcare_utilization.py

```diff
@@ -243,6 +287,50 @@ def on_time_step_cleanup(self, event: Event) -> None:
             ]
         )
 
+    def test_fpg(self, pop_visitors: pd.DataFrame) -> pd.Index:
+        not_already_enrolled = pop_visitors[data_values.COLUMNS.LIFESTYLE].isna()
+        bmi = self.bmi(pop_visitors.index)
+        age = pop_visitors["age"]
+
+        fpg_not_tested_recently = (
+            # never been tested for FPG
+            pop_visitors[data_values.COLUMNS.LAST_FPG_TEST_DATE].isna()
+        ) | (
+            # last FPG test more than 3 years ago
+            pop_visitors[data_values.COLUMNS.LAST_FPG_TEST_DATE]
+            < self.clock()
+            - pd.Timedelta(days=365.25 * data_values.FPG_TESTING.MIN_YEARS_BETWEEN_TESTS)
+        )
+
+        is_eligible_for_testing = (
+            (not_already_enrolled)
+            & (age >= data_values.FPG_TESTING.AGE_ELIGIBILITY_THRESHOLD)
+            & (bmi >= data_values.FPG_TESTING.BMI_ELIGIBILITY_THRESHOLD)
+            & (fpg_not_tested_recently)
+        )
+
+        # Determine which simulants eligible for FPG testing actually get tested
+        tested_simulants = self.randomness.filter_for_probability(
+            pop_visitors[is_eligible_for_testing],
+            [data_values.FPG_TESTING.PROBABILITY_OF_TESTING_GIVEN_ELIGIBLE]
+            * sum(is_eligible_for_testing),
+        )
+
+        # Assign FPG test date to current time
+        pop_visitors.loc[
+            tested_simulants.index, data_values.COLUMNS.LAST_FPG_TEST_DATE
+        ] = self.clock()
+
+        self.population_view.update(
```

Comment:
Unless it's unavoidable, it's better to have a single call to update a population view at the end of the `on_timestep_cleanup` function. This `test_fpg` function could just return a mask of simulants who should get tested like `determine_followups_sbp` does.